HW-13/main.py

```diff
@@ -0,0 +1,141 @@
+    if command_splitted[0] == 'delete' and len(command_splitted) > 1:
+        splitted_value = command.split(" ", 1)[1]
+        my_list = phonebook.keys()
+        if splitted_value in my_list:
+            del phonebook[splitted_value]
+            with open('phonebook.json', 'w+') as file:
+                json.dump(phonebook, file)
+            print(f'{splitted_value} successfully deleted')
+            print(f'Now you have {len(phonebook)} contacts')
+            print(phonebook)
+            command = ''
+            continue
+        else:
+            print(f'There is no contact {splitted_value}')
+            command = ''
+            continue
+    elif command_splitted[0] == 'delete':
+        print("All data has been deleted")
+        phonebook.clear()
+        with open('phonebook.json', 'w') as file:
+            json.dump(phonebook, file)
+        command = ''
+
+    # ------------------------------------------------------------------------------------------
+
+    command_splitted = command.split(" ", 1)
+    if command_splitted[0] == 'show' and len(command_splitted) > 1:
+        splitted_value = command.split(" ", 1)[1]
+        my_list = phonebook.keys()
+        if splitted_value in my_list:
+            print(f'Here is full info about user {splitted_value}: {phonebook.get(splitted_value)}')
+            command = ''
+            continue
+        else:
+            print(f'There is no contact {splitted_value}')
+            command = ''
+            continue
+    elif command_splitted[0] == 'show':
+        answer = input('Please enter name to show: ')
+        if answer in phonebook.keys():
+            print(f'Here is full info about user {answer}: {phonebook.get(answer)}')
+        command = ''
+
+    # ------------------------------------------------------------------------------------------
+
+    if command == 'list':
+        if len(phonebook) == 0:
+            print('Phonebook is empty')
+        else:
+            for key in phonebook:
+                print(key)
+                command = ''
+            continue
+        command = ''
+
+    # ------------------------------------------------------------------------------------------
+
+    if command == 'exit':
+        break
+
+
+# 3. (необов'язкове виконання) Написати програму, яка буде:
+# зчитувати текст із файлу, назва якого вводиться користувачем (program argument або input)
+# знаходити всі email в тексті і змінювати їх на X***@****X, де замість Х мають бути перша і остання літери справжньої адреси, а весь інший текст має бути замінений на *. Кількість * необовʼязково має відповідати кількості замінених символів
+
+
+def replace_emails(file_name):
+    with open(file_name, "r") as file:
+        text = file.read()
+    email_regex = r"[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+"
+    emails = re.findall(email_regex, text)
+    for email in emails:
+        first_char = email[0]
+        last_char = email[-1]
+        middle = "*" * (len(email) - 2)
```

Comment:
Too many `*` in the replaces email. This is not a mistake according to the task, but you did it wrong if you wanted to replace the email address with the amount of `*` that equals email's symbols count